src/modules/solution-explorer/solution-explorer-list/solution-explorer-list.ts

```diff
@@ -138,7 +138,12 @@ export class SolutionExplorerList {
     } catch (error) {
       this._solutionService.removeSolutionEntryByUri(uri);
 
-      return;
+      /**
+       * TODO: The error message only contains 'Failed to fetch' if the connection
+       * failed. A more detailed cause (such as Connection Refused) would
+       * be better. This needs to be implemented in the Service or Repository.
+       */
+      throw new Error('Failed to retrieve the list of ProcessModels from the endpoint');
```

Comment:
would use `receive` instead of retrieve
```suggestion
      throw new Error('Failed to receive the list of ProcessModels from the endpoint');
```